sealtk/core/AbstractProxyModel.cpp

```diff
@@ -67,10 +67,25 @@ bool AbstractProxyModel::lessThan(
   switch (role)
   {
     // String comparisons
-    case core::NameRole:
     case core::UniqueIdentityRole:
       return localeAwareLessThan(left.toString(), right.toString());
 
+    case core::NameRole:
+    {
+      bool leftIsInt, rightIsInt;
```

Comment:
I'm inclined to instead follow [how Qt does](https://github.com/qt/qtbase/blob/c69b218197e97f09bf69f8f7fa3b4677c0f17455/src/corelib/itemmodels/qabstractitemmodel.cpp#L552-L587) this. To wit, something more like:
```c++
  case core::NameRole:
    switch (left.type())
    {
      case QVariant::Int:
      case QVariant::LongLong:
        return left.toLongLong() < right.toLongLong();
      case QVariant::UInt:
      case QVariant::ULongLong:
        return left.toULongLong() < right.toULongLong();
      default:
        return localeAwareLessThan(left.toString(), right.toString());
    }
```

(I wouldn't bother with more types than this — name should either be a number or a string, not strange things like dates — but importantly, ID's are `qint64`, not `int`.)

Reply:
I like this better -- Done.